### PR TITLE
Add new JMX pack

### PR DESCRIPTION
**Pull request by Kami**
This pull request adds a JMX pack. Initial version of the pack contains a sensor which allows users to monitor Java applications for different attributes and metrics exposed via JMX protocol.

This allows users to consume attributes and monitor many popular Java applications such as Cassandra, Tomcat, Hadoop and many others.

In the future, I also plan to add a JMX action which will allow users to manipulate JMX attributes and invoke methods exposed via JMX. Support for invoking methods should be quite powerful since many services such as Cassandra expose all of the management operations (and more) which are available via CLI also via JMX (in fact, CLI uses JMX underneath).

Note: This pull request also brings up a known limitation again - currently we have a no concept of "config template" and "config instance" (or something along those lines) which means user can only run one instance of a sensor and only monitor one JMX service and attribute (same limitation exists for many other existing packs such as jira).

### CODE REVIEW

**Comment by jfryman** [comment]
> Note: This pull request also brings up a known limitation again - currently we have a no concept of "config template" and "config instance" (or something along those lines) which means user can only run one instance of a sensor and only monitor one JMX service and attribute (same limitation exists for many other existing packs such as jira).

+:100:

File: packs/jmx/extern/jmxquery/jmxquery/JMXQuery.java [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+  private ArrayList<String> attributeKeys = new ArrayList<String>();;
+  private String object;
+  private String username, password;
+
+  private static int maxMatchingObjects = 15;
+
+  private Object checkData;
+
+  private void connect() throws IOException
+  {
```
Comment (manasdk) [inline packs/jmx/extern/jmxquery/jmxquery/JMXQuery.java]: brace on previous line :)

Comment (Kami) [inline packs/jmx/extern/jmxquery/jmxquery/JMXQuery.java]: Heh, I actually imported this from ck-agent repo unmodified.

It's true that I wrote it, but the changes I did were mostly just modifications and additions to an existing code from jmxquery repo and I just followed an existing style back then :P

Comment (manasdk) [inline packs/jmx/extern/jmxquery/jmxquery/JMXQuery.java]: Alright. Wrote java far too long ... the reaction was almost instinctive.

Comment (Kami) [inline packs/jmx/extern/jmxquery/jmxquery/JMXQuery.java]: Heh, though so.

**Comment by manasdk** [comment]
> Note: This pull request also brings up a known limitation again - currently we have a no concept of "config template" and "config instance" (or something along those lines) which means user can only run one instance of a sensor and only monitor one JMX service and attribute (same limitation exists for many other existing packs such as jira).

+1

Yes we should definitely get it done soon.

**Comment by DoriftoShoes** [comment]
+1 HA for sensor containers is definitely a concern at CITEIS.  If we get support for this in the platform that would be awesome.

**Pull request merged**